File: apps/mybookkeeper/frontend/src/shared/types/lease/signed-lease-update-request.ts
import type { SignedLeaseStatus } from "@/shared/types/lease/signed-lease-status";

export interface SignedLeaseUpdateRequest {
  notes?: string | null;
  status?: SignedLeaseStatus;
  values?: Record<string, unknown>;
  /** Per-lease toggle for the auto-email-tenant-on-generate behavior. */
  auto_email_tenant?: boolean;
}

File: apps/mybookkeeper/backend/app/schemas/leases/signed_lease_update_request.py
"""Schema for PATCH /signed-leases/{id} — notes, status, values (draft-only)."""
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, ConfigDict


class SignedLeaseUpdateRequest(BaseModel):
    notes: str | None = None
    status: str | None = None
    # Only honoured when current status == "draft" — service enforces.
    values: dict[str, Any] | None = None
    # Per-lease toggle for the auto-email-tenant-on-generate feature.
    # ``None`` (omitted) leaves the existing value alone; ``True`` /
    # ``False`` overwrites it.
    auto_email_tenant: bool | None = None

    model_config = ConfigDict(extra="forbid")
